crates/router/src/connector/braintree/transformers.rs

```diff
@@ -255,234 +544,1268 @@ impl From<BraintreePaymentStatus> for enums::AttemptStatus {
+        let query = VOID_TRANSACTION_MUTATION.to_string();
+        let variables = VariableCancelInput {
+            input: CancelInputData {
+                transaction_id: item.request.connector_transaction_id.clone(),
+            },
+        };
+        Ok(Self { query, variables })
+    }
+}
+
+#[derive(Debug, Clone, Deserialize, Serialize)]
+pub struct CancelResponseTransactionBody {
+    id: String,
+    status: BraintreePaymentStatus,
+}
+
+#[derive(Debug, Clone, Deserialize, Serialize)]
+pub struct CancelTransactionData {
+    reversal: CancelResponseTransactionBody,
+}
+
+#[derive(Debug, Clone, Deserialize, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct CancelResponseData {
+    reverse_transaction: CancelTransactionData,
+}
+
+#[derive(Debug, Clone, Deserialize, Serialize)]
+pub struct CancelResponse {
+    data: CancelResponseData,
+}
+
+#[derive(Debug, Clone, Deserialize, Serialize)]
+#[serde(untagged)]
+pub enum BraintreeCancelResponse {
+    CancelResponse(Box<CancelResponse>),
+    ErrorResponse(Box<ErrorResponse>),
+}
+
+impl<F, T>
+    TryFrom<types::ResponseRouterData<F, BraintreeCancelResponse, T, types::PaymentsResponseData>>
+    for types::RouterData<F, T, types::PaymentsResponseData>
 {
     type Error = error_stack::Report<errors::ConnectorError>;
     fn try_from(
-        item: types::RefundsResponseRouterData<api::Execute, RefundResponse>,
+        item: types::ResponseRouterData<F, BraintreeCancelResponse, T, types::PaymentsResponseData>,
     ) -> Result<Self, Self::Error> {
-        Ok(Self {
-            response: Ok(types::RefundsResponseData {
-                connector_refund_id: item.response.id,
-                refund_status: enums::RefundStatus::from(item.response.status),
+        match item.response {
+            BraintreeCancelResponse::ErrorResponse(error_response) => Ok(Self {
+                response: build_error_response(&error_response.errors, item.http_code),
+                ..item.data
             }),
-            ..item.data
+            BraintreeCancelResponse::CancelResponse(void_response) => {
+                let void_data = void_response.data.reverse_transaction.reversal;
+                let status = enums::AttemptStatus::from(void_data.status.clone());
+                let response = if utils::is_payment_failure(status) {
+                    Err(types::ErrorResponse {
+                        code: void_data.status.to_string().clone(),
+                        message: void_data.status.to_string().clone(),
+                        reason: None,
+                        attempt_status: None,
+                        connector_transaction_id: Some(void_data.id),
+                        status_code: item.http_code,
+                    })
+                } else {
+                    Ok(types::PaymentsResponseData::TransactionResponse {
+                        resource_id: types::ResponseId::ConnectorTransactionId(void_data.id),
```

Comment:
Do we need to update the transaction_id after void?

Reply:
its returning the same transactionId (Not needed) 
removed